modules/core/keeper/keeper.go

```diff
@@ -76,3 +84,17 @@ func (k *Keeper) SetRouter(rtr *porttypes.Router) {
 	k.Router = rtr
 	k.Router.Seal()
 }
+
+func checkEmptyKeepers(stakingKeeper clienttypes.StakingKeeper, upgradeKeeper clienttypes.UpgradeKeeper,
```

Comment:
I think it might be nicer not to abstract these into a function and just do the checks directly within `NewKeeper`. 

I agree with the approach of checking if the keeper is empty as opposed to if the params are initialized :+1: